File: src/main/java/com/rajan/puparulesengine/rulesinterpreter/rule/condition/Condition.java
package com.rajan.puparulesengine.rulesinterpreter.rule.condition;

import com.fasterxml.jackson.databind.JsonNode;
import com.rajan.puparulesengine.constant.enums.ConditionTypeEnum;
import com.rajan.puparulesengine.constant.enums.OperatorEnum;
import lombok.Data;

import java.util.List;

@Data
public abstract class Condition {
    ConditionTypeEnum type;
    String field;
    OperatorEnum operator;
    String value;
    List<Condition> conditions; // only populated for complex conditions

    public abstract boolean evaluate(JsonNode node);
}

File: frontend/src/app/rules-engine/model/Condition.ts
export class Condition {
  // values: orcondition, andcondition, simplecondition
  type: string;
  name: string;

  // field set for orcondition, andcondition
  conditions: Condition[];

  // simplecondition properties
  field: string;
  operator: string;
  value: string;
}
